src/stringobj.c

```diff
@@ -198,17 +198,17 @@ void LoadChar( Obj c )
 
 /****************************************************************************
 **
-*F  FuncEmptyString( <self>, <len> ) . . . . . . . empty string with space
-*
-* Returns an empty string, but with space for len characters preallocated.
-*
+*F  FuncEmptyString( <self>, <len> ) . . . . . . . .  empty string with space
+**
+**  Returns an empty string, but with space for len characters preallocated.
+**
 */
 Obj    FuncEmptyString( Obj self, Obj len )
 {
     Obj                 new;
-    while ( ! IS_INTOBJ(len) ) {
+    while ( ! IS_NONNEG_INTOBJ(len) ) {
```

Comment:
Strictly speaking this is a bug fix preventing OOB access. Perhaps should be mentioned in the commit message. Not sure if we care about it for release notes - perhaps a generic entry "improved input validation for several kernel functions"?

Reply:
I think some generic comment like that would be useful.